protected/controllers/BaseAdminController.php

```diff
@@ -209,17 +209,26 @@ protected function genericAdmin($title, $model, array $options = array(), $key =
 
                         $to_delete = $model::model()->findAll($criteria);
                         foreach ($to_delete as $i=>$item) {
-                            if (!$item->delete()) {
-                                $tx->rollback();
-                                $error = $item->getErrors();
-                                foreach ($error as $e)
-                                {
-                                    $errors[$i]=$e[0];
+                            try {
+                                if (!$item->delete()) {
+                                    $tx->rollback();
+                                    $error = $item->getErrors();
+                                    foreach ($error as $e) {
+                                        $errors[$i] = $e[0];
+                                    }
+                                    Yii::app()->user->setFlash('error.error', implode('<br/>', $errors));
+                                    $this->redirect(Yii::app()->request->url);
                                 }
-
-                                Yii::app()->user->setFlash('error.error', implode('<br/>', $errors));
+                            } catch(Exception $excep){
+                                switch ($excep->getCode()){
+                                    case 23000:
+                                        $message = "Deleting Error: disorder group '".$item->name."' is in use";
+                                        break;
+                                    default:
+                                        $message = "Operation failed due to error ".$excep->getCode();
```

Comment:
I think we should direct users to contact Support if the generic error handling is executed. At the moment a user that sees this error will have no idea what to do.